### PR TITLE
Refresh readme

### PR DESCRIPTION
**Pull request by koivunej**
Mostly about highlighting the use of docker, but also:
- make it more user friendly by not repeating the commands
- add newlines around code blocks
- remove transient docker examples (always mount volume)
- remove non-functioning parts with `path-to-pathfinder`

Rendered [README.md](https://github.com/eqlabs/pathfinder/blob/59a5c170b944260ada2847fdcf781bbdf23122f0/README.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,6 +244,7 @@ The following environment variables can be passed to the container:
 
 ### Building the container image yourself
 
+Building the container image from source code is necessary on in special cases or development.
```
Comment (CHr15F0x) [inline README.md:247]:
```suggestion
Building the container image from source code is necessary only in special cases or development.
```

**Review by CHr15F0x — APPROVED** [review]
LGTM, just one typo (?)

File: README.md [hunk 2]
```diff
@@ -56,28 +62,30 @@ rustup update
 `pathfinder` requires Python version `3.7` or `3.8`. (In particular, `cairo-lang` 0.7.1 seems incompatible with Python 3.10.)
 
 ```bash
-sudo apt install python3
-sudo apt install python3-venv
-sudo apt install python3-dev
+sudo apt install python3 python3-venv python3-dev
```
[the remark this answers is not in the document]
Comment (koivunej) [reply README.md:65]: UX when doing these multiple times has abysmal worst case with multiple password prompts.

**Review by kkovaacs — APPROVED** [review]
LGTM

**Review by Mirko-von-Leipzig — APPROVED** [review]
Thanks!

**Pull request merged**